service_configuration_lib/spark_config.py

```diff
@@ -1089,12 +1089,17 @@ def get_spark_conf(
             # We want to make the app name more unique so that we can search it
             # from history server.
             app_name = f'{app_base_name}_{ui_port}_{int(time.time())}'
+        is_jupyter = _is_jupyterhub_job(app_name)
 
         # Explicitly setting app id: replace special characters to '_' to make it consistent
         # in all places for metric systems:
         # - since in the Promehteus metrics endpoint those will be converted to '_'
         # - while the 'spark-app-selector' executor pod label will keep the original app id
-        app_id = re.sub(r'[\.,-]', '_', app_name)
+        if is_jupyter:
+            raw_app_id = app_name
+        else:
+            raw_app_id = f'{paasta_service}_{paasta_instance}_{int(time.time())}'
```

Comment:
```suggestion
            raw_app_id = f'{paasta_service}__{paasta_instance}__{int(time.time())}'
```

Reply:
I was thinking that we only have 53 character available, this will make more app ids being trimmed. What would you think? Both are okay to me

Reply:
I saw from Tron web, there are already quite a few servce+instance names > 53 chars, even if the length of timestamp is not counted in

Reply:
I don't think 2 characters matter here much, when trimming happens. But having double underscores gives us the value of separating instance and service names. Can app.id contain '.'? If so, you can set it as <service>.<instance>
Is the timestamp here in millis or seconds? I would have have timestamp in seconds (even every 10second is fine to save a digit but that can be confusing) or even "timestamp % 100,000 (or 10,000)" as it's mostly for uniqueness. 

Thoughts?

Reply:
We are replacing special characters `.`, `,`, `-` to `_` to make app_id consistent with Prometheus metric names.
The timestamp is in seconds. Modding the timestamp sounds good to me